### PR TITLE
release: add config to trigger SBOM creation

### PR DESCRIPTION
**Pull request by samsalisbury**
This should cause CRT to trigger the [crt-generate-sbom](https://github.com/hashicorp/security-generate-release-sbom/blob/main/.github/workflows/crt-generate-sbom.yml) workflow post-release. Currently that workflow just stores the SBOM as a GitHub Actions Workflow Artifact. The purpose of this change is to establish the integration between Vault and Vault Enterprise releases and our SBOM-generating workflow. Once this link is established, we will continue to work on the SBOM side of things and shouldn't need to make further changes here.

See [PSP-1992](https://hashicorp.atlassian.net/browse/PSP-1992) and related tickets for more details.

**Question for reviewer:** do I need to make a similar change in Vault Enterprise, or will this be automatically ported there as well? Thanks.

[PSP-1992]: https://hashicorp.atlassian.net/browse/PSP-1992?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by ryancragun — APPROVED** [review]
This looks good to me. This change will be backported but I suspect it'll result in a merge conflict that needs a manual resolution. I'll be happy to update/merge/resolve that for you if you think it's good to go.

**Comment by samsalisbury** [comment]
Thanks @ryancragun, I can't merge this myself, if you're happy with it would you be able to merge? Thanks.

**Pull request merged**
